### PR TITLE
mmseqs2 clustering code improvements

### PR DESCRIPTION
**Pull request by YoelShoshan**

### CODE REVIEW

File: fusedrug/data/protein/ops/loaders/fasta_loader.py [hunk 1]
```diff
@@ -1,25 +1,53 @@
-import pyfastx  # https://pyfastx.readthedocs.io/en/latest/usage.html
+# import pyfastx  # https://pyfastx.readthedocs.io/en/latest/usage.html
```
Comment (mosheraboh) [inline fusedrug/data/protein/ops/loaders/fasta_loader.py:1]: remove

File: fusedrug/utils/cluster/cluster_using_mmseqs.py [hunk 2]
```diff
@@ -149,24 +159,52 @@ def cluter_impl(
...
 
     print(
         "B.3 - generate cluster TSV for convinience"
     )  # for massive datasets, we might skip this and use mmseqs output format directly (possibly worth checking if there's already a python lib that handles this)
-    clustered_tsv = os.path.join(output_dir, "clustered.tsv")
+    clustered_tsv = join(output_dir, "clustered.tsv")
     cmd = f"mmseqs createtsv {step_B_initial_db} {step_B_initial_db} {clustered_db} {clustered_tsv}"
     _run_system_cmd(cmd)
 
+    ####
```
Comment (mosheraboh) [inline fusedrug/utils/cluster/cluster_using_mmseqs.py:174]: What is this new step? is it relevant to ab design as well?

File: fusedrug/utils/file_formats/indexed_fasta_custom.py [hunk 3]
```diff
@@ -2,29 +2,35 @@
 from fuse.utils.cpu_profiling import Timer
 from fuse.utils.file_io import save_hdf5_safe, load_hdf5, change_extension
 import numpy as np
-from fuse.utils.misc.context import DummyContext
+
+# from fuse.utils.misc.context import DummyContext
```
Comment (mosheraboh) [inline fusedrug/utils/file_formats/indexed_fasta_custom.py:6]: remove

**Review by mosheraboh — APPROVED** [review]
LGTM.
minors inline as usual

File: fusedrug/utils/cluster/cluster_using_mmseqs.py (hunk 2 above)
Comment (YoelShoshan) [reply fusedrug/utils/cluster/cluster_using_mmseqs.py:174]: the point of this new step is that a FASTA file is generated that contains the clusters centers ("representatives") including their sequence

if you want to train an AB model using only the centers, it's relevant, as you can easily get them and their respective sequence in a single FASTA file

**Pull request merged**